sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -411,9 +450,173 @@ def tune(
                                 env_from=env_from if env_from else None,
                                 resources=resources_per_trial,
                             )
-                        ],
+            
+            pod_spec = client.V1PodTemplateSpec(
+                metadata=models.V1ObjectMeta(
+                    annotations={"sidecar.istio.io/inject": "false"}
+                ),
+                spec=client.V1PodSpec(
+                    restart_policy="Never",
+                    containers=[container_spec],
+                ),
+            )
+
+        # If users choose to use external models and datasets.
+        else:
+            try:
+                from kubeflow.storage_initializer.hugging_face import \
+                    HuggingFaceDatasetParams
+                from kubeflow.storage_initializer.hugging_face import \
+                    HuggingFaceModelParams
+                from kubeflow.storage_initializer.hugging_face import \
+                    HuggingFaceTrainerParams
+                from kubeflow.storage_initializer.s3 import S3DatasetParams
+                import peft
+                import transformers
+            except ImportError:
+                raise ImportError(
+                    "Tune API dependencies not installed. "
+                    + "Run: pip install -U 'kubeflow-training[huggingface]' "
+                )
+            
+            # Create PVC for the Storage Initializer.
+            try:
+                self.core_api.create_namespaced_persistent_volume_claim(
+                    namespace=namespace,
+                    body=utils.get_pvc_spec(
+                        pvc_name=constants.STORAGE_INITIALIZER,
```

Comment:
Should we just have pvc_name == experiment_name, in that case every Katib Experiment will have a separate PVC.
I know that, we don't do this for Training Operator, but we should fix it there as well.
WDYT @johnugeorge @helenxie-bit @deepanker13 ?

Reply:
@andreyvelich I think it will be better than using a constant pvc name. I will make adjustment here. Should I also fix it in Training Operator?

Reply:
Yes, please submit PR to change it in the Training Operator as well!

Reply:
@andreyvelich I have made adjustments here and also submitted a PR to the Training Operator. Please review it when you have time: https://github.com/kubeflow/training-operator/pull/2187